src/main/java/bio/terra/janitor/db/JanitorDao.java

```diff
@@ -88,6 +87,18 @@ public void createResource(TrackedResource resource, Map<String, String> labels)
         DataAccessUtils.singleResult(jdbcTemplate.query(sql, params, TRACKED_RESOURCE_ROW_MAPPER)));
   }
 
+  /** Returns the {@link TrackedResource} for a {@link CloudResourceUid} if there is one. */
+  @Transactional(propagation = Propagation.SUPPORTS)
+  public Optional<TrackedResource> retrieveTrackedResource(CloudResourceUid cloudResourceUid) {
+    String sql =
+        "SELECT id, resource_uid, creation, expiration, state FROM tracked_resource tr "
+            + "WHERE resource_uid::jsonb = :resource_uid::jsonb";
+    MapSqlParameterSource params =
+        new MapSqlParameterSource().addValue("resource_uid", serialize(cloudResourceUid));
+    return Optional.ofNullable(
+        DataAccessUtils.singleResult(jdbcTemplate.query(sql, params, TRACKED_RESOURCE_ROW_MAPPER)));
```

Comment:
this is ok for this PR, if you add a TODO to remove it later.
We don't want this long term because there's no database guarantee that there will be at most one result for this query.